src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -46,6 +55,26 @@ def make_install_req_from_link(link, parent):
     )
 
 
+def make_install_req_from_editable(link, parent):
+    # type: (Link, InstallRequirement) -> InstallRequirement
+    assert parent.editable, "parent not editable"
+    return install_req_from_editable(
+        link.url,
+        # HACK: install_req_from_editable accepts Optional[str] here, but
+        # parent.comes_from is Union[str, InstallRequirement, None]. How do
+        # we fix the type hint conflicts?
+        comes_from=parent.comes_from,  # type: ignore
```

Comment:
@pradyunsg Should we fix the type hints in `install_req_from_editable`?

Reply:
Go ahead!